onnxruntime/core/session/inference_session.h

```diff
@@ -691,6 +691,7 @@ class InferenceSession {
   mutable onnxruntime::OrtMutex session_mutex_;  // to ensure only one thread can invoke Load/Initialize
   bool is_model_loaded_ = false;                 // GUARDED_BY(session_mutex_)
   bool is_inited_ = false;                       // GUARDED_BY(session_mutex_)
+  bool is_concurrent_run_supported_ = true;      // Graph execution in Run is GUARDED_BY(session_mutex_) if false
```

Comment:
Not only XNNPack has the restriction. DML, and probably many other 3rd-party EPs have it too.

Reply:
So probably we should let EP tell us this point?